# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsParent.cpp
# Commit: 0287f5604a69
# Full Hash: 0287f5604a69c65705743257a5bbe24a5bdb9d9f
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-12-14 23:03:37
# Regressor Bug: 1678030
# File Overlap Count: 1
# Description:
#   Bug 1678030 - Move CollectWhileHasResult to QuotaCommon and use in other quota clients. r=dom-workers-and-storage-reviewers,ttung
#   
#   This patch also:
#   
#   - introduces CollectElementsWhileHasResult(Typed) to collects elements in and
# ==============================================================================

diff -r 6dd8d66f8116 -r 0287f5604a69 dom/indexedDB/ActorsParent.cpp
--- a/dom/indexedDB/ActorsParent.cpp	Thu Dec 10 11:45:42 2020 +0000
+++ b/dom/indexedDB/ActorsParent.cpp	Mon Dec 14 10:42:36 2020 +0000
@@ -685,14 +685,6 @@
   return NS_OK;
 }
 
-template <typename StepFunc>
-Result<Ok, nsresult> CollectWhileHasResult(mozIStorageStatement& aStmt,
-                                           StepFunc&& aStepFunc) {
-  return CollectWhile(
-      [&aStmt] { IDB_TRY_RETURN(MOZ_TO_RESULT_INVOKE(aStmt, ExecuteStep)); },
-      [&aStmt, &aStepFunc] { return aStepFunc(aStmt); });
-}
-
 nsresult SetJournalMode(mozIStorageConnection& aConnection) {
   MOZ_ASSERT(!NS_IsMainThread());
 